UniConnectApp/app/src/main/java/com/example/uniconnect/MainViewModel.kt

```diff
@@ -32,40 +33,42 @@ class MainViewModel(var universityService: IUniversityService = UniversityServic
 
     fun listenToThisUserPost() {
         user?.let { user ->
-            firestore.collection("users").document(user.uid).collection("posts").addSnapshotListener { snapshot, error ->
-                // see of we received an error
-                if (error != null) {
-                    Log.w("listen failed.", error)
-                    return@addSnapshotListener
-                }
-                // if we reached this point, there was not an error, and we have data.
-                snapshot?.let {
-                    val allPosts = ArrayList<Post>()
+            firestore.collection("users").document(user.uid).collection("posts")
+                .addSnapshotListener { snapshot, error ->
+                    // see of we received an error
+                    if (error != null) {
+                        Log.w("listen failed.", error)
+                        return@addSnapshotListener
+                    }
+                    // if we reached this point, there was not an error, and we have data.
+                    snapshot?.let {
+                        val allPosts = ArrayList<Post>()
 
-                    /*allPosts.add(Post(NEW_SPECIMEN))*/
+                        /*allPosts.add(Post(NEW_SPECIMEN))*/
```

Comment:
As long as you're here, remove this entirely.
Avoid pushing commented code to Version Control.  Version control has tools that makes this unnecessary.
- You can look at history to see what used to be there.
- You can use branches for experimental features.